src/main/java/uk/gov/companieshouse/api/accounts/controller/PreviousPeriodController.java

```diff
@@ -92,6 +91,50 @@ public ResponseEntity create(@Valid @RequestBody PreviousPeriod previousPeriod,
         return responseEntity;
     }
 
+    @PutMapping
+    public ResponseEntity update(@RequestBody @Valid PreviousPeriod previousPeriod,
+            BindingResult bindingResult, @PathVariable("companyAccountId") String companyAccountId,
+            HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.PREVIOUS_PERIOD.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        Errors errors = new Errors();
+
+        if (bindingResult.hasErrors()) {
+            errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+        }
+
+        previousPeriodValidator.validatePreviousPeriod(previousPeriod);
```

Comment:
The errors returned from validatePreviousPeriod are not being checked?

Reply:
This was copy and pasted from current period ... but yes I will add the error assignment